doc/specs/application-auth-token.md

```diff
@@ -48,17 +116,45 @@ The proper way to sign the token is as follows:
 3. Sign with ed25519 ECDSA
 4. HexEncode the result bytes into a string
 
-```text
-AAT {
+```bash
+  AAT = {
     ApplicationSignature: "",
     ApplicationPublicKey: a.ApplicationPublicKey,
     ClientPublicKey:      a.ClientPublicKey,
     Version:              a.Version,
-}
+  }
+  AATBytes = JSON.Encode(AAT)
+  Message = SHA3_256(AATBytes)
+  AAT.ApplicationSignature = ED25519.Sign(Message)
 ```
 
-`AATBytes = JSON.Encode(AAT)`
+### Relay Generation
 
-`Message = SHA3-256(AATBytes)`
+The `Client` is needed to sign the actual relays while the `Application` gets
+charged on-chain. However, the `Application` only gets charged if the `Client`
+signed the relay. The `Client` public key is in the `AAT` and had to be signed
+by the `Application` during `AAT` generation.
 
-`AAT.Signature = ED25519.Sign(Message)`
+```mermaid
+flowchart TB
+    RP([Relay Proof])
+    CG([ClientPrivateKey/ \n GatePrivateKey])
+
+    subgraph RPU["Unsigned RelayProof"]
+        requestHash
+        Entropy
+        SessionHeight
+        AAT
+        ServicerPubKey
+        BlockchainID
+    end
+
+    AAT <-.-> CG
```

Comment:
This link should be removed from this diagram IMO: it makes the diagram harder to understand as it looks to create a cycle. AAT generation is presented a bit earlier on this page.